Add cached tables documentation (#7880)
pingcap/docs: sql-statements/sql-statement-alter-table.md

|   ( 'WITH' | 'WITHOUT' ) 'VALIDATION'
|   'SECONDARY_LOAD'
|   'SECONDARY_UNLOAD'

> |   ( 'AUTO_INCREMENT' | 'AUTO_ID_CACHE' | 'AUTO_RANDOM_BASE' | 'SHARD_ROW_ID_BITS' ) EqOpt LengthNum

Author: Yes. To keep it consistent with the Chinese version: https://github.com/pingcap/docs-cn/blob/master/sql-statements/sql-statement-alter-table.md

* Spatial data types are not supported.

> * `ALTER TABLE t CACHE | NOCACHE` is not a standard MySQL syntax but a TiDB extension. See [Cached Tables](/cached-tables.md) for more information.

Reviewer: SQL is a standard, but for MySQL it is only a reference implementation. I think we can improve the readability by just calling it an extension to MySQL syntax:
```suggestion
* `ALTER TABLE t CACHE | NOCACHE` is a TiDB extension to MySQL syntax. See [Cached Tables](/cached-tables.md) for more information.
```
